demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,582 @@
+                        if (FLAGS_r)
+                            rawOutputHeadpose(i, out_y_fc[i], out_p_fc[i], out_r_fc[i]);
+                    }
+
+                    if (!FLAGS_m_lm.empty()) {
+                        landmarksDataUpdate(face, out_landmarks[i]);
+                        if (FLAGS_r)
+                            rawOutputLandmarks(i, out_landmarks[i]);
+                    }
+                    // End of face postprocesing
+
+                    faces.push_back(face);
+                }
+
+                presenter->drawGraphs(frame);
+
+                //  Visualizing results
+                if (!FLAGS_no_show || !FLAGS_o.empty()) {
+                    out.str("");
+                    out << "Total image throughput: " << std::fixed << std::setprecision(2)
+                        << 1000.f / (timer["total"].getSmoothedDuration()) << " fps";
+                    cv::putText(frame, out.str(), THROUGHPUT_METRIC_POSITION, cv::FONT_HERSHEY_TRIPLEX, 1,
+                                cv::Scalar(255, 0, 0), 2);
+
+                    // drawing faces
+                    visualizer->draw(frame, faces);
+
+                    cv::imshow("Detection results", frame);
+
+                    int key = cv::waitKey(1);
+                    if (27 == key || 'Q' == key || 'q' == key) {
+                        stream.stop();
+                    } else {
+                        presenter->handleKey(key);
+                    }
+                }
+
+                if (!FLAGS_o.empty() && !videoWriter.isOpened()) {
+                    videoWriter.open(FLAGS_o, cv::VideoWriter::fourcc('I', 'Y', 'U', 'V'), 25, cv::Size(frame.size()));
+                }
+                if (!FLAGS_o.empty()) {
+                    videoWriter.write(frame);
+                }
+
+                framesCounter++;
+                timer.finish("total");
+            } else { // End of streaming
+                if(FLAGS_loop_video) {
+                    slog::info << "Number of processed frames: " << framesCounter << slog::endl;
+                    slog::info << "Total image throughput: " << framesCounter * (1000.f / timer["total"].getTotalDuration()) << " fps" << slog::endl;
+
+                    std::cout << presenter->reportMeans() << '\n';
+
+                    slog::info << "Setting media source" << slog::endl;
+                    try {
+                        if (FLAGS_i == "cam")
+                            stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(0));
+                        else
+                            stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(FLAGS_i));
+                    } catch (const std::exception& error) {
+                        std::stringstream msg;
+                        msg << "Can't open source {" << FLAGS_i << "}" << std::endl;
+                        throw std::invalid_argument(msg.str());
+                    }
```

Comment:
This code is repeated too.

Reply:
Reduced the code to a `setInput()` call
Is that ok?

Reply:
It's a good direction, but the logging and error handling could go into that function too.

Reply:
WIP

Reply:
Got rid of duplicate code
Used construction construction as

`do {
    ...
} while(FLAGS_loop);`

Not sure if this is a good solution

Reply:
Seems fine by me. 👍